ui/components-react/src/components-react/filter-builder/FilterBuilderRule.scss

```diff
@@ -2,44 +2,70 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
+.fb-row {
+  --background-color: var(--iui-color-background);
+  --border-color: var(--iui-color-border);
+  --operator-width: 5rem;
+  --property-name-width: 25%;
+  --condition-width: 10rem;
+  --toolbar-width: 6.25rem;
+  border: 1px solid var(--iui-color-border);
 
-.rule {
-  display: grid;
-  grid-template-columns: min-content auto;
-  padding: 3px 0;
+  display: block;
+  position: relative;
 
-  .rule-condition {
-    display: grid;
-    grid-template-columns: minmax(auto, 500px) 150px minmax(auto, 100%);
-    grid-column-gap: var(--iui-size-m);
-    width: calc(100% - var(--iui-size-s));
+  > div {
+    background: var(--background-color);
+    align-items: stretch;
+    overflow: hidden;
+  }
 
-    .iui-input-container.iui-with-message {
-      .rule-value {
-        grid-area: inputs;
-      }
-    }
+  .fb-row-container {
+    padding-block: var(--iui-size-xs);
+    padding-inline: var(--iui-size-m);
+    gap: var(--iui-size-m);
   }
 
-  .rule-remove-action {
-    padding-right: var(--iui-size-s);
-    visibility: hidden;
+  &[data-iui-size="small"] .fb-row-container {
+    padding-block: var(--iui-size-2xs);
   }
 
-  &:focus-within .rule-remove-action,
-  &:hover .rule-remove-action {
-    visibility: visible;
+  .fb-row-container {
+    flex: 1;
+  }
+
+  .fb-row-operator {
+    width: var(--operator-width);
+    border-right: 1px solid var(--border-color);
+  }
+
+  .fb-row-name {
+    width: var(--property-name-width);
   }
-}
 
-.rule-group-items > .rule {
-  padding-bottom: 14px;
+  .fb-row-condition {
+    width: var(--condition-width);
+  }
 
-  &:first-child {
-    padding-top: 14px;
+  .fb-row-value {
+    flex: 1;
+    width: 15rem;
   }
 
-  &:only-child {
-    padding-top: 3px;
+  .fb-row-toolbar {
+    width: var(--toolbar-width);
+    margin-left: auto;
   }
 }
+
+.fb-row:not(:last-child) {
+  //only show buttons on last row to prevent visual clutter
+  .iui-button:first-child {
```

Comment:
Avoid referencing iTwinUI internal class names. The preferred way is to add your class name to the component and select based on that.